common/arch/ogl/ogl_extensions.cpp

```diff
@@ -73,16 +97,33 @@ static support_mode is_supported(const char *extensions, const array<long, 2> &v
 
 void ogl_extensions_init()
 {
-	const char *version_str = reinterpret_cast<const char *>(glGetString(GL_VERSION));
+	const auto *version_str = reinterpret_cast<const char *>(glGetString(GL_VERSION));
 	if (!version_str) {
 		con_printf(CON_URGENT, "no valid OpenGL context when querying GL extensions!");
 		return;
 	}
 	const auto version = parse_version_str(version_str);
-	const char *extension_str = reinterpret_cast<const char *>(glGetString(GL_EXTENSIONS));
+	const auto extension_str = reinterpret_cast<const char *>(glGetString(GL_EXTENSIONS));
+
+	con_printf(CON_VERBOSE, "OpenGL%s: version %ld.%ld (%s)",
+#if DXX_USE_OGLES
```

Comment:
According to `info cpp` node *Directives Within Macro Arguments*:
```
Occasionally it is convenient to use preprocessor directives within the
arguments of a macro.  The C and C++ standards declare that behavior in
these cases is undefined.
```
`con_printf` is defined as a macro to enable some checking on its usage.  All supported gcc versions handle this as you intended.  clang on Linux handles it.  I cannot test clang on OS X.  Regardless, it would produce slightly simpler runtime code to restructure this as:
```c++
#if DXX_USE_OGLES
#define DXX_OGL_STRING	" ES"
#else
#define DXX_OGL_STRING	""
	con_printf(CON_VERBOSE, "OpenGL" DXX_OGL_STRING ": version %ld.%ld (%s)", version[0], version[1], version_str);
#undef DXX_OGL_STRING
```
This ensures that string pasting, but no preprocessor directives, occur within the macro arguments.